CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/damage/MCDamageSource.java

```diff
@@ -16,99 +13,130 @@
 		return entity.getInternal() instanceof EntityLivingBase ? source.getDeathMessage((EntityLivingBase)entity.getInternal()).getFormattedText() : "ERROR";
 	}
+
 	@Override
 	public float getHungerDamage() {
 		return source.getHungerDamage();
 	}
+
+	@Override
+	public float getHungerDamageFixed() {
+		return source.getHungerDamage();
+	}
+
 	@Override
 	public IEntity getImmediateSource() {
-		Entity ent = source.getImmediateSource();
-		if (ent instanceof EntityPlayer) return new MCPlayer((EntityPlayer) ent);
-		if (ent instanceof EntityLivingBase) return new MCEntityLivingBase((EntityLivingBase) ent);
-		return new MCEntity(ent);
+		return CraftTweakerMC.getIEntity(source.getImmediateSource());
 	}
+
 	@Override
 	public IEntity getTrueSource() {
-		Entity ent = source.getTrueSource();
-		if (ent instanceof EntityPlayer) return new MCPlayer((EntityPlayer) ent);
-		if (ent instanceof EntityLivingBase) return new MCEntityLivingBase((EntityLivingBase) ent);
-		return new MCEntity(ent);
+		return CraftTweakerMC.getIEntity(source.getTrueSource());
 	}
+
 	@Override
 	public boolean isCreativePlayer() {
 		return source.isCreativePlayer();
 	}
+
 	@Override
 	public boolean isDamageAbsolute() {
 		return source.isDamageAbsolute();
 	}
+
+	@Override
+	public boolean isDamageUnblockable() {
+		return source.isUnblockable();
+	}
+
 	@Override
 	public boolean isDifficultyScaled() {
 		return source.isDamageAbsolute();
 	}
+
 	@Override
 	public boolean isExplosion() {
 		return source.isExplosion();
 	}
+
 	@Override
 	public boolean isFireDamage() {
 		return source.isFireDamage();
 	}
+
 	@Override
 	public boolean isMagicDamage() {
 		return source.isMagicDamage();
 	}
+
 	@Override
 	public boolean isProjectile() {
 		return source.isProjectile();
 	}
+
+	@Override
+	@Nullable
+	public IVector3d getDamageLocation() {
+		return CraftTweakerMC.getIVector3d(source.getDamageLocation());
+	}
+
 	@Override
 	public IDamageSource setDamageAllowedInCreativeMode() {
 		return new MCDamageSource(source.setDamageAllowedInCreativeMode());
 	}
+
 	@Override
 	public IDamageSource setDamageBypassesArmor() {
 		return new MCDamageSource(source.setDamageBypassesArmor());
 	}
+
 	@Override
 	public IDamageSource setDamageIsAbsolute() {
 		return new MCDamageSource(source.setDamageIsAbsolute());
 	}
+
 	@Override
 	public IDamageSource setDifficultyScaled() {
 		return new MCDamageSource(source.setDifficultyScaled());
 	}
+
 	@Override
 	public IDamageSource setExplosion() {
 		return new MCDamageSource(source.setExplosion());
 	}
+
 	@Override
 	public IDamageSource setFireDamage() {
 		return new MCDamageSource(source.setFireDamage());
 	}
+
 	@Override
 	public IDamageSource setMagicDamage() {
 		return new MCDamageSource(source.setMagicDamage());
 	}
+
 	@Override
 	public IDamageSource setProjectile() {
 		return new MCDamageSource(source.setProjectile());
 	}
-    
-    @Override
+
+	@Override
     public Object getInternal() {
         return source;
     }

```

Comment:
indentation is weird, I know it wasn't your PR, but if you're changing the file could you please fix it